src/main/java/com/salesforce/cdp/queryservice/util/TokenHelper.java

```diff
@@ -236,6 +239,17 @@ public static CoreTokenRenewResponse getCoreToken(String url, String refreshToke
         return coreTokenRenewResponse;
     }
 
+    private static Token exchangeTokenWithRefresh(String url, String refreshToken,
+                                                  String coreToken, String clientId, String secret,
+                                                  String dataspace,OkHttpClient client) throws TokenException {
+        try{
+            return exchangeToken(url,coreToken,dataspace,client);
+        } catch (TokenException e) {
+            log.error("Core token is not valid fetching a new token ");
```

Comment:
Can we please do `log.error("Core token is not valid fetching a new token ", e);` here

Reply:
Done added it .